### ISSUE TITLE
Warning on props description when props got with destructuring and has passed trough an object 

### ISSUE THREAD

**Issue by budarin** [eslint/eslint#9770]
Have this code:
```js
type ComputeMatchTypeT = {
    computedMatch?: MatchT,
    location?: LocationT,
    path?: string,
    strict?: boolean,
    exact?: boolean,
    sensitive?: boolean,
};
...
    static computeMatch(props: ComputeMatchTypeT, context: LocactionContextT): MatchT | true {
        const { computedMatch, location, path, strict, exact, sensitive } = props;

        if (computedMatch) {
            return computedMatch;
        }

        const pathname = (location || context.location).pathname;

        return (path && matchPath(pathname, { path, strict, exact, sensitive })) || true;
    }
```
Got such warnings: `ESLint: 'sensitive' PropType is defined but prop is never used (react/no-unused-prop-types)`
This warning presents on props: computedMatch, path, strict, exact, sensitive - because they are not used directly like location.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. It looks like this warning is coming from the `react/no-unused-prop-types` rule, which is part of [`eslint-plugin-react`](https://github.com/yannickcr/eslint-plugin-react). If you have a question/bug in that rule, I would recommend reporting it on that repository instead.
